lib/services/docker.pm

```diff
@@ -0,0 +1,53 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Package for cups service tests
+#
+# Maintainer: qa-c team <qa-c@suse.de>
+
+package services::docker;
+use base 'opensusebasetest';
+use testapi;
+use utils;
+use strict;
+use warnings;
+use containers::docker;
+use containers::utils 'basic_container_tests';
+use containers::container_images 'build_and_run_image';
+
+my $service_type = 'Systemd';
+
+# check service is running and enabled
+sub check_service {
+    common_service_action 'docker', $service_type, 'is-enabled';
+    common_service_action 'docker', $service_type, 'is-active';
+}
+
+# check docker service before and after migration
+# stage is 'before' or 'after' system migration.
+sub full_docker_check {
+    my (%hash) = @_;
+    my ($stage, $type) = ($hash{stage}, $hash{service_type});
+    $service_type = $type;
+    my $engine = containers::docker->new();
+    $engine->init() if ($stage eq 'before');
```

Comment:
there is a duplication here with line following from https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/edbae20345112c136837ad3ea49ea03f8820d8fe/lib/service_check.pm#L283 and below.

```
zypper_call "in $srv_pkg_name";
common_service_action($srv_proc_name, $service_type, 'enable');
common_service_action($srv_proc_name, $service_type, 'start');
common_service_action($srv_proc_name, $service_type, 'is-active');
```
IIUC `full_docker_check` will be executed before those lines (https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/edbae20345112c136837ad3ea49ea03f8820d8fe/lib/service_check.pm#L280) and pretty much will do the same things.
- install docker
- will check if is-active

Reply:
Well I would like to keep my changes to minimum. I'm still trying to solve earlier solution so let's wait for verification run.

Reply:
i dont think it makes sense. if `$engine` is not initialized the rest of the commands will fail later. i cant say what the solution would be here and i cant say more without dig into the code and the test. But i think my observation is obvious.